.github/workflows/project_ci.yml

```diff
@@ -0,0 +1,91 @@
+# In YAML, 'raw' and 'endraw' can't come at the beginning of a token or at the end of a quoted value, which is why they're in weird places. They are
+# needed to escape YAML templating variables, which still allowing cookiecutter to replace the value of the repository name.
+
+# Instructions to clear caches:
+# 1) List all caches for this repo using the Github CLI: gh api  -H "Accept: application/vnd.github+json" /repos/Lightmatter/django-hydra/actions/caches
+# 2) Delete each cache id from the list, e.g.: gh api --method DELETE  -H "Accept: application/vnd.github+json" /repos/Lightmatter/django-hydra/actions/caches/1
+
+name: Project CI
+on: pull_request
+
+jobs:
+  create_and_test_project:
+    name: Create and test Django project
+    runs-on: ubuntu-latest
+    env:
+      DATABASE_URL: "psql://postgres:postgres@localhost/postgres"
+      DJANGO_SECRET_KEY: "!!!! Change me !!!!"
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Install and configure Poetry
+        uses: snok/install-poetry@e3dbfd357f4751d4f582d62bc8a71e56c2a7015b # v1.3.2
+        with:
+          version: 1.2.1
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
+
+      # Configure Python for the template
+      - uses: actions/setup-python@v4
+        id: setup_python
+        with:
+          python-version-file: ".python-version"
+          cache: "poetry"
+
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v3
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
+
+      - name: Install Hydra dependencies
+        run: |
+          poetry env use ${{ steps.setup_python.outputs.python-version }}
+          poetry install --no-interaction --no-root
+
+      # Instantiate the project
+      - name: Instantiate project using the values in cookiecutter.json
+        run: poetry run cookiecutter . --no-input --accept-hooks no
+
+      # Configure the project environment
+      - uses: actions/setup-node@v3
+        with:
+          node-version: lts/*
+          cache: "npm"
+          cache-dependency-path: "**/package-lock.json"
+
+      - name: Install NPM package
+        working-directory: ./sampleapp
+        run: npm install
+
+      - name: Install Sample App dependencies
+        working-directory: ./sampleapp
+        run: |
+          poetry env use ${{ steps.setup_python.outputs.python-version }}
```

Comment:
Doesn't this need to be done with the version of Python being used by the project, and not by the template?

Reply:
Good point. Though I hate that this is just manually specified here, cause it's gonna change then this run will be using an incorrect version.

Reply:
Yeah fair. The other issue is that going back to manually specifying the other version of Python might break the changes you made to caching.  (Not sure, I just found it to be not entirely intuitive in terms of being able to predict what would and wouldn't work.)